src/AasxServerStandardBib/Services-DB/AssetAdministrationShellServiceDB.cs

```diff
@@ -0,0 +1,561 @@
+                throw new($"Submodel with id {submodelIdentifier} NOT found in AAS with id {aasIdentifier}");
+            }
+        }
+
+        public void DeleteSubmodelById(string aasIdentifier, string submodelIdentifier)
+        {
+            var found = IsSubmodelPresentWithinAAS(aasIdentifier, submodelIdentifier);
+            if (found)
+            {
+                _logger.LogDebug($"Found submodel with id {submodelIdentifier} in AAS with id {aasIdentifier}");
+                //delete the submodel first, this should eventually delete the submodel reference from all the AASs
+                _submodelService.DeleteSubmodelById(submodelIdentifier);
+            }
+            else
+            {
+                throw new($"Submodel with id {submodelIdentifier} NOT found in AAS with id {aasIdentifier}");
+            }
+        }
+
+        public void DeleteSubmodelElementByPath(string aasIdentifier, string submodelIdentifier, string idShortPath)
+        {
+            var found = IsSubmodelPresentWithinAAS(aasIdentifier, submodelIdentifier);
+            if (found)
+            {
+                _logger.LogDebug($"Found submodel with id {submodelIdentifier} in AAS with id {aasIdentifier}");
+                _submodelService.DeleteSubmodelElementByPath(submodelIdentifier, idShortPath);
+            }
+            else
+            {
+                throw new($"Submodel with id {submodelIdentifier} NOT found in AAS with id {aasIdentifier}");
+            }
+
+        }
+
+
+
+        public void DeleteSubmodelReferenceById(string aasIdentifier, string submodelIdentifier)
+        {
+            var aas = _packageEnvService.GetAssetAdministrationShellById(aasIdentifier, out _);
+
+            if (aas != null)
+            {
+                var submodelReference = aas.Submodels.Where(s => s.Matches(submodelIdentifier));
+                if (submodelReference.Any())
+                {
+                    _logger.LogDebug($"Found requested submodel reference in the aas.");
+                    bool deleted = aas.Submodels.Remove(submodelReference.First());
+                    if (deleted)
+                    {
+                        _logger.LogDebug($"Deleted submodel reference with id {submodelIdentifier} from the AAS with id {aasIdentifier}.");
+                        Program.signalNewData(1);
+                    }
+                    else
+                    {
+                        _logger.LogError($"Could not delete submodel reference with id {submodelIdentifier} from the AAS with id {aasIdentifier}.");
+                    }
+                }
+                else
+                {
+                    throw new NotFoundException($"SubmodelReference with id {submodelIdentifier} not found in AAS with id {aasIdentifier}");
+                }
+            }
+        }
+
+
+
+        public List<IAssetAdministrationShell> GetAllAssetAdministrationShells(List<SpecificAssetId> assetIds = null, string idShort = null)
```

Comment:
Consider using LINQ for filtering to enhance readability and maintainability within this method.